tests/sentry/api/endpoints/test_organization_sentry_function_details.py

```diff
@@ -0,0 +1,113 @@
+from unittest.mock import patch
+
+from django.urls import reverse
+
+from sentry.testutils import APITestCase
+from sentry.testutils.helpers import Feature
+
+
+class OrganizationSentryFunctionDetails(APITestCase):
+    endpoint = "sentry-api-0-organization-sentry-function-details"
+
+    def setUp(self):
+        super().setUp()
+        self.create_organization(owner=self.user, name="RowdyTiger")
+        self.login_as(user=self.user)
+
+    @patch("sentry.api.endpoints.organization_sentry_function.create_function")
+    def test_get_valid_function(self, mock_func):
+        creation_endpoint = reverse(
+            "sentry-api-0-organization-sentry-functions", args=[self.organization.slug]
+        )
+        with Feature("organizations:sentry-functions"):
+            creation_response = self.client.post(
+                creation_endpoint,
+                data={"name": "foo", "author": "bar", "code": "baz", "overview": "qux"},
+            )
+            assert creation_response.status_code == 201
+            mock_func.assert_called_once_with("baz", creation_response.data["external_id"], "qux")
+            get_function_endpoint = reverse(
+                self.endpoint, args=[self.organization.slug, creation_response.data["slug"]]
+            )
+            get_response = self.client.get(get_function_endpoint)
+            assert get_response.status_code == 200
+            assert creation_response.data == get_response.data
+
+    def test_get_invalid_function(self):
+        get_function_endpoint = reverse(
+            self.endpoint, args=[self.organization.slug, "invalid-slug"]
+        )
+        get_response = self.client.get(get_function_endpoint)
+        assert get_response.status_code == 404
+
+    @patch("sentry.api.endpoints.organization_sentry_function.create_function")
+    @patch("sentry.api.endpoints.organization_sentry_function_details.update_function")
+    def test_edit_valid_function(self, mock_update_func, mock_create_func):
+        creation_endpoint = reverse(
+            "sentry-api-0-organization-sentry-functions", args=[self.organization.slug]
+        )
+        with Feature("organizations:sentry-functions"):
+            creation_response = self.client.post(
```

Comment:
@vpandiarajan20 you can always put the creation logic in a `setUp` method for this class so it always happens before the test